AppServer/google/appengine/tools/appengine_rpc.py

```diff
@@ -611,7 +617,8 @@ def Send(self, request_path, payload="",
           else:
             raise
     finally:
-      socket.setdefaulttimeout(old_timeout)
+      if not opener_has_timeout:
+        socket.setdefaulttimeout(old_timeout)
```

Comment:
@cdonati @jeanleonov could old_timeout be initialized down here since its not used before this point?